lectures/12-plonk.tex

```diff
@@ -0,0 +1,234 @@
+\(W : \Omega \rightarrow \Omega\) is a permutation of \(\Omega\) if \(\forall i \in [k], W(\omega^i) = \omega^{j}\) is a bijection.
+\end{definition}
+
+\begin{example}
+\text{Example } $(k=3): \quad W(\omega^0) = \omega^2, \quad W(\omega^1) = \omega^0, \quad W(\omega^2) = \omega^1$
+\end{example}
+
+Let \(f\) and \(g\) be polynomials in \(\mathbb{F}_p^{(\leq d)}[X]\). Verifier has commitments \([f]\), \([g]\), \([W]\).
+
+\underline{Goal:} prover wants to prove that \(f(y) = g(W(y))\) for all \(y \in \Omega\), equivalent to \(g(\Omega)\) is the same as \(f(\Omega)\), permuted by the prescribed \(W\).
+
+Naive way of doing this is by running \textbf{ZeroTest} on \(f(y) - g(W(y)) = 0\) on \(\Omega\), however that would result in proving needing to manipulate polynomials of degree \(k^2\), resulting in quadratic prover time. We can reduce this to a product-check on a polynomial of degree \(2k\).
+
+\begin{remark}
+Observation: If \((W(\alpha), f(\alpha))_{\alpha \in \Omega}\) is a permutation of \((\alpha, g(\alpha))_{\alpha \in \Omega}\) then \(f(y) = g(W(y))\) for all \(y \in \Omega \).
+\end{remark}
+
+\begin{example}
+\[
+\text{Proof by example: } W(\omega^0) = \omega^2, \quad W(\omega^1) = \omega^0, \quad W(\omega^2) = \omega^1 
+\]
+\[
+\text{Right tuple: } (\omega^0, g(\omega^0)), (\omega^1, g(\omega^1)), (\omega^2, g(\omega^2))
+\]
+\[
+\text{Left tuple: } (\omega^2, f(\omega^0)), (\omega^0, f(\omega^1)), (\omega^1, f(\omega^2))
+\]
+\end{example}
+
+Let \(\hat{f}(X,Y) = \prod_{\alpha \in \Omega} (X - Y \cdot W(\alpha) - f(\alpha))\) and
+\(\hat{g}(X,Y) = \prod_{a \in \Omega} (X - Y \cdot a - g(a))\) (bivariate polynomials of total degree \(k\)).
+
+\begin{lemma}
+\(\hat{f}(X,Y) = \hat{g}(X,Y) \iff \left( W(\alpha), f(\alpha) \right)_{\alpha \in \Omega}\) is a permutation of \(\left( \alpha, g(\alpha) \right)_{\alpha \in \Omega}\).
+
+To prove, use the fact that \(\mathbb{F}_p[X,Y]\) is a unique factorization domain.
+\end{lemma}
+
+\paragraph{The complete protocol.}
+\begin{enumerate}
+    \item Verifier generates random \(r, g \leftarrow\mathbb{F}_p^{(\leq d)} \)
+    \item Run \textbf{ProductCheck} on \(\hat{f}(r, s) = \hat{g}(r, s): \prod_{a \in \Omega} \left( \frac{r - s \cdot W(a) - f(a)}{r - s \cdot a - g(a)} \right) = 1\)
+\end{enumerate}
+
+This would imply that \(\hat{f}(X,Y) = \hat{g}(X,Y)\) with high probability due to Schwartz-Zippel.
+
+\begin{remark}
+Complete and sound, assuming \(2d/p\) is negligible.
+\end{remark}
+
+\subsection{Plonk Arithmetization}
+
+Assume that we have a certain arithmetic circuit \(C\) with a \(|C|\) number of gates and \(|I|\ = |I_x| + |I_w|\) number of inputs. We encode this circuit, recording its computation trace in a table, where rows represent the state per each gate, while columns are of the form \((a, b, c)\), where \(a\) and \(b\) are left and right inputs, and \(c\) is the output of the gate. In this manner, the output of the last gate corresponds to the output of the circuit.
```

Comment:
You might keep the notation as it is, but in the previous section, I used $\mathcal{I}$ to denote the set of inputs ($\mathcal{I}\_{\text{io}}$ for public ones, while $\mathcal{I}\_{\text{mid}}$ for intermediate witness components). Also, as far as I remember, I used circuit letter C with `\texttt`.